apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -426,28 +473,58 @@ def _cs_reader(
 class SparseTensorReadIterBase(somacore.ReadIter[_RT], metaclass=abc.ABCMeta):
     """Private implementation class"""
 
-    def __init__(self, sr: clib.SOMAArray, shape: NTuple):
-        self.sr = sr
+    def __init__(
+        self,
+        array: SparseNDArray,
+        coords: options.SparseDFCoords,
+        shape: NTuple,
+        result_order: clib.ResultOrder,
+        platform_config: Optional[options.PlatformConfig],
+    ):
+        self.array = array
+        self.coords = coords
         self.shape = shape
+        self.result_order = result_order
+        self.platform_config = platform_config
+
+        sr = array._handle._handle
```

Comment:
```suggestion
        clib_handle = array._handle._handle
```
(the 'old name' `sr` is now confusing)